src/state/routing/useRoutes.ts

```diff
@@ -0,0 +1,74 @@
+import { Currency, CurrencyAmount, Ether, Token } from '@uniswap/sdk-core'
+import { FeeAmount, Pool, Route } from '@uniswap/v3-sdk'
+import { useMemo } from 'react'
+import { GetQuoteResult } from 'state/routing/slice'
+
+export function useRoutes(
```

Comment:
can you explain this function in a comment? also seems like it doesn't need to be a hook unless you want to use the token info from token lists to get the logos

Reply:
```
/**
 * Transforms a Routing API quote into an array of routes that
 * can be used to create a V3 `Trade`.
 */
```

I made it a hook so it would memoize itself rather than having the caller memoize it. Do you prefer if I keep this as a function and have the caller memoize? I guess a regular function is easier to test but all the tests are already written with `renderHook`

Reply:
i would rather it be a function and the caller does the memoize